kv/replica_slice.go

```diff
@@ -103,7 +105,8 @@ func (rs replicaSlice) SortByCommonAttributePrefix(attrs []string) int {
 		return 0
 	}
 	topIndex := len(rs) - 1
-	for bucket := 0; bucket < len(attrs); bucket++ {
+	bucket := 0
```

Comment:
could you back out this part of the change (doesn't seem like an improvement) or explain why you prefer it this way?
Also, remove the TODO on line 102.